src/prefect/tasks.py

```diff
@@ -342,6 +342,17 @@ def __init__(
             )
         self.on_completion = on_completion
         self.on_failure = on_failure
+
+        if not (
+            callable(retry_condition_fn)
+            or is_async_fn(retry_condition_fn)
+            or retry_condition_fn is None
+        ):
```

Comment:
@dominictarro I think we can just check if it's not callable. We don't want the other conditions to raise an error.

Reply:
Not totally understanding. I can drop the `is_async_fn` since that doesn't matter until down the line and is covered by `callable`, but I still need to check if it's a callable or None. If `retry_condition_fn` is neither, should it not raise an error? What is the check doing if not erroring on improper argument types?

Reply:
ah I think I reviewed this too quickly yesterday, my apologies. I see now that `retry_condition_fn` being `None` wouldn't raise an error here, so we're aligned.

Reply:
@dominictarro I find it a tad hard to read as it is, though. 

Maybe something like:

```python
is_callable = callable(retry_condition_fn)
is_none = retry_condition_fn is None

if not (is_callable or is_none):
    raise
```

Reply:
or
```python
if retry_condition_fn is not None and not(callable(retry_condition_fn)):
```

Reply:
Yeah, sounds good. I can make it more legible

Reply:
Thank you! 🙌